src/v/datalake/coordinator/coordinator.cc

```diff
@@ -249,19 +249,31 @@ checked<ss::gate::holder, coordinator::errc> coordinator::maybe_gate() {
     return gate_.hold();
 }
 
+struct coordinator::table_schema_provider {
```

Comment:
nit: just a thought, this file is getting a bit crowded to read through with the different implementations of this schema provider sprinkled throughout. Might be easier on the eyes to separate it out into its own header and cc

Reply:
I kind of like it being close to the respective functions as the only reason for existence of this classes is to factor out differences in those functions